mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/dao/DaoUpdateMethodGenerator.java

```diff
@@ -0,0 +1,241 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.internal.mapper.processor.dao;
+
+import static com.datastax.oss.driver.internal.mapper.processor.dao.ReturnTypeKind.FUTURE_OF_ASYNC_RESULT_SET;
+import static com.datastax.oss.driver.internal.mapper.processor.dao.ReturnTypeKind.FUTURE_OF_VOID;
+import static com.datastax.oss.driver.internal.mapper.processor.dao.ReturnTypeKind.RESULT_SET;
+import static com.datastax.oss.driver.internal.mapper.processor.dao.ReturnTypeKind.VOID;
+
+import com.datastax.oss.driver.api.core.cql.BoundStatement;
+import com.datastax.oss.driver.api.core.cql.BoundStatementBuilder;
+import com.datastax.oss.driver.api.core.cql.SimpleStatement;
+import com.datastax.oss.driver.api.mapper.annotations.Update;
+import com.datastax.oss.driver.api.querybuilder.QueryBuilder;
+import com.datastax.oss.driver.internal.core.util.concurrent.CompletableFutures;
+import com.datastax.oss.driver.internal.mapper.processor.ProcessorContext;
+import com.datastax.oss.driver.internal.mapper.processor.util.generation.GeneratedCodePatterns;
+import com.datastax.oss.driver.internal.querybuilder.DefaultRaw;
+import com.datastax.oss.driver.internal.querybuilder.update.DefaultUpdate;
+import com.datastax.oss.driver.shaded.guava.common.annotations.VisibleForTesting;
+import com.squareup.javapoet.ClassName;
+import com.squareup.javapoet.MethodSpec;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.element.VariableElement;
+
+public class DaoUpdateMethodGenerator extends DaoMethodGenerator {
+
+  private static final EnumSet<ReturnTypeKind> SUPPORTED_RETURN_TYPES =
+      EnumSet.of(VOID, FUTURE_OF_VOID, RESULT_SET, FUTURE_OF_ASYNC_RESULT_SET);
+  private static final Pattern USING_TIMESTAMP_VALUE_PATTER =
+      Pattern.compile("(?i)USING TIMESTAMP (\\d+$)");
+  private static final Pattern USING_TIMESTAMP_BIND_MARKER_PATTERN =
+      Pattern.compile("(?i)USING TIMESTAMP :(\\w+$)");
+  private static final Pattern USING_TTL_VALUE_PATTER = Pattern.compile("(?i)USING TTL (\\d+)$");
+  private static final Pattern USING_TTL_BIND_MARKER_PATTERN =
+      Pattern.compile("(?i)USING TTL :(\\w+)$");
+  // todo should we support ? bindMarker
```

Comment:
I am not sure if we should support `USING TTL ?` syntax in the `customUsingClause`. 
I think we don't need it since we are binding parameters from the method arguments using named parameters

Reply:
As we discussed offline, let's support named parameters only.

However this makes me realize something else: unlike `Insert`, we can't simply append the provided string at the end of the query (because USING is not at the end for UPDATE, which is quite an unfortunate choice by the CQL language designers tbh).

Hence the string parsing you have to do here; it's completely justified but I find it a bit awkward. So let's revisit the annotation syntax to have the user provide the values directly instead:
```
@Update(ttl = "...", timestamp = "...")
```
Both attributes are strings that default to empty.
* if the string is empty we don't generate the corresponding clause
* if it starts with `:` it's a bind marker name
* otherwise we try to parse it as an int (for ttl) or long (for timestamp). If parsing fails we issue a compiler error.

I'm creating a ticket to implement a similar change for `@Insert`.

Reply:
One more thing 😃 
With 2257, concatenating won't work anymore for `@Insert.customWhereClause` (because USING comes after). We'll have to use `whereRaw()` (like you initially did on this PR), and therefore it's simpler if we don't require the `WHERE` keyword in the provided string.
Same for `customIfClause` (no `IF` keyword in the string, use `ifRaw` to concatenate).

Again I'm creating a ticket to revisit existing annotations ([JAVA-2258](https://datastax-oss.atlassian.net/browse/JAVA-2258)), do it the new way directly in this PR.

Reply:
I fixed 2257 on the base branch. You should be able to reuse the two protected methods in `DaoMethodGenerator`.

Reply:
I see one potential problem with `ifRaw`  -> I think that there could be a problem with creating the enclosed ifs:
So the following clause: "IF a = a AND b =b" - previously we would just append code. With `ifRaw()` you will need to create two `ifRaw()` statements (I didn't test that yet)

Reply:
`ifRaw` appends whatever you pass as-is, so if you put the `AND` keyword in your string it should work fine.
```java
@Update(customIfClause = "a = :a AND b = :b"
```